src/entropic/sinkhorn_stabilized.jl

```diff
@@ -1,161 +1,160 @@
     end
 
-    _scaling_factor = scaling_factor === nothing ? 1//2 : scaling_factor
-    _scaling_steps = scaling_steps === nothing ? 5 : scaling_steps
-
-    # initial regularization parameter
-    εstep = ε * _scaling_factor^(1 - _scaling_steps)
-    @debug "ε-scaling Sinkhorn algorithm: ε = $εstep"
+    return nothing
+end
 
-    # initialize cache and perform stabilized Sinkhorn algorithm
-    cache = SinkhornStabilizedCache(μ, ν, C, εstep)
-    solve!(cache, μ, ν, C, εstep; kwargs...)
+# absorption step
+function absorb!(solver::SinkhornSolver{<:SinkhornStabilized})
+    # unpack
+    ε = solver.eps
+    cache = solver.cache
+    u = cache.u
+    v = cache.v
+    α = cache.alpha
+    β = cache.beta
 
-    for _ in 2:(_scaling_steps - 1)
-        εstep *= _scaling_factor
-        @debug "ε-scaling Sinkhorn algorithm: ε = $εstep"
+    # update dual potentials
+    α .+= ε .* log.(u)
+    β .+= ε .* log.(v)
 
-        # re-run Sinkhorn algorithm with smaller regularization parameter
-        updateKtilde!(cache, C, εstep)
-        solve!(cache, μ, ν, C, εstep; kwargs...)
-    end
+    # reset `α` and `β`
+    fill!(u, one(eltype(u)))
+    fill!(v, one(eltype(v)))
 
-    # re-run Sinkhorn algorithm with desired regularization parameter
-    @debug "ε-scaling Sinkhorn algorithm: ε = $ε"
-    updateKtilde!(cache, C, ε)
-    solve!(cache, μ, ν, C, ε; kwargs...)
+    # update kernel
+    update_K!(solver)
 
-    # compute final plan
-    updateKtilde!(cache, C, ε)
-    γ = cache.Ktilde
+    return nothing
+end
 
-    return γ
+# update kernel
+function update_K!(solver::SinkhornSolver{<:SinkhornStabilized})
+    cache = solver.cache
+    α = add_singleton(cache.alpha, Val(2))
+    β = add_singleton(cache.beta, Val(1))
+    @. cache.K = exp(-(solver.C - α - β) / solver.eps)
+    return nothing
 end
 
-"""
-    sinkhorn_stabilized(
-        μ, ν, C, ε;
-        absorb_tol=1_000,
-        maxiter=1_000,
-        atol=0,
-        rtol=atol > 0 ? 0 : √eps,
-        check_convergence=10,
-        maxiter=1_000,
-    )
+# obtain plan
+function plan(solver::SinkhornSolver{<:SinkhornStabilized})
+    absorb!(solver)
+    return copy(solver.cache.K)
+end
 
-Compute the optimal transport plan for the entropically regularized optimal transport
-problem with source and target marginals `μ` and `ν`, cost matrix `C` of size
-`(length(μ), length(ν))`, and entropic regularisation parameter `ε`.
+# deprecations
 
-The function uses the log-domain stabilized Sinkhorn algorithm with an absorption tolerance
-`absorb_tol`.
+"""
+    sinkhorn_stabilized(μ, ν, C, ε; absorb_tol=1_000, kwargs...)
 
-Every `check_convergence` steps it is assessed if the algorithm is converged by checking if
-the iterate of the transport plan `G` satisfies
+This method is deprecated, please use
 ```julia
-isapprox(sum(G; dims=2), μ; atol=atol, rtol=rtol, norm=x -> norm(x, 1))
+sinkhorn(
+    SinkhornStabilized(; absorb_tol=absorb_tol), μ, ν, C, ε; kwargs...
```

Comment:
is the algorithm the first argument? Isn't it the last? I mean `sinkhorn(μ, ν, C, ε, SinkhornStabilized(; absorb_tol=absorb_tol)`.

Reply:
Ah yeah, initially I had it as the first argument but then switched it around. Seems I forgot to update the documentation, good catch!

Reply:
Fixed.